service/src/client.rs

```diff
@@ -68,9 +69,26 @@ impl<S: PostService> ServiceClient<S> {
         tls: Option<(Option<String>, Certificate, Identity)>,
         service: S,
     ) -> eyre::Result<Self> {
-        let endpoint = Channel::builder(address.parse()?)
-            .keep_alive_timeout(Duration::from_secs(20))
-            .http2_keep_alive_interval(Duration::from_secs(10 * 60));
+        let listen_address = address.parse::<Uri>()?;
+        let parts = listen_address.clone().into_parts();
+        let scheme = listen_address.scheme_str();
+        let scheme = match scheme {
+            Some("http") | Some("https") => scheme,
+            None => Some("http"),
+            _ => return Err(eyre::eyre!("unknown client protocol")),
+        }
+        .unwrap();
```

Comment:
I think it should also reject an empty scheme (the `None` variant), shouldn't it? 

How about simplifying and avoiding cloning (the scheme is available in `parts` already):
```suggestion
        let parts = listen_address.into_parts();
        let scheme = parts.scheme.ok_or(eyre::eyre!("unknown client protocol"))?;
        if !["http", "https"].contains(&scheme.as_str()) {
            return Err(eyre::eyre!("invalid scheme: {scheme}"));
        }
```

Note: you will also need to change:

```rust
                match scheme {
                    "https" => {
```
to 
```rust
                match scheme.as_str() {
                    "https" => {
```

Reply:
Fallback for empty variant was set to conform second option of requested:
> services should NOT accept address when a http/https is not specified directly OR default to http
which is giving a bit of backward compatibility.

Thx for the suggested simplification, I'll use that!